util/oc_process.c

```diff
@@ -366,6 +366,38 @@ oc_process_is_closing_all_tls_sessions(void)
   return false;
 }
 #endif /* OC_SECURITY */
+/*---------------------------------------------------------------------------*/
+int
+oc_process_drop(const struct oc_process *p, oc_process_drop_event_t drop_event,
+                void *user_data)
+{
+  int dropped = 0;
+
+  if (!p || !drop_event) {
+    return dropped;
+  }
+
+  oc_process_num_events_t i = 0;
+  while (i < nevents) {
+    oc_process_num_events_t index = (fevent + i) % OC_PROCESS_NUMEVENTS;
+    const struct event_data *event = &events[index];
+    if (event->p != p || !drop_event(event->ev, event->data, user_data)) {
+      // move to the next event
+      ++i;
+      continue;
+    }
+    // we have a match, drop the event and move the last event to its position
+    if (nevents > 1) {
+      oc_process_num_events_t last_index =
+        (fevent + nevents - 1) % OC_PROCESS_NUMEVENTS;
+      events[index] = events[last_index];
```

Comment:
I am wondering whether changing the order of messages is dangerous or not.  From reading the headers I see no mention of a guaranteed order of execution. What do you think, won't we get some weird bugs?

Reply:
This is a issue for TCP connection - stream can be broken.

Reply:
I see, oh well. I guess we have to move the memory and keep the order - should be  0-2 memset calls (might be annoying to get the bounds and size right for the circular buffer) or a for-loop which moves all the elements forward by one.

Reply:
Fixed via memmove :)

Reply:
Looks good, tested it on paper with one case and the bounds were correct.